src/backend/core/api/serializers.py

```diff
@@ -228,5 +230,24 @@ class InvitationSerializer(serializers.ModelSerializer):
 
     class Meta:
         model = models.Invitation
-        fields = ["email", "team", "role", "issuer"]
-        read_only_fields = ["team", "issuer"]
+        fields = ["id", "created_at", "email", "team", "role", "issuer", "is_expired"]
+        read_only_fields = ["id", "created_at", "team", "issuer", "is_expired"]
+
+    def validate(self, attrs):
+        existing_identity = models.Identity.objects.filter(email=attrs["email"])
+
+        # verify invitee has not already joined the app / the team
+        if existing_identity:
+            existing_access = models.TeamAccess.objects.filter(
+                team=self.initial_data["team"]
+            )
+            if existing_access:
+                raise serializers.ValidationError(
+                    "This member already joined your team."
+                )
+
+            raise serializers.ValidationError(
+                _("This email is already associated to a registered user.")
+            )
+
+        return attrs
```

Comment:
nit early return to avoid nesting. I am a bit scared to have introduced regression, is this logic identical with your original code?

Also, I have added `.exists()` to the end of filters; Which should be a bit more performant:

> simply reads a single record in [the most optimized way](https://github.com/django/django/blob/b996424199cca58f64b77faeaa367b7e7efb9776/django/db/models/sql/query.py#L528):


```suggestion
    def validate(self, attrs):
    
        existing_identity = models.Identity.objects.filter(email=attrs["email"]).exists()
        
        # Verify invitee has not already joined the app / the team
        if not existing_identity:
                return attrs
                
        existing_access = models.TeamAccess.objects.filter(
                team=self.initial_data["team"]
        ).exists()
        
        if existing_access:
                raise serializers.ValidationError(
                        "This member already joined your team."
                )

        raise serializers.ValidationError(
                _("This email is already associated to a registered user.")
        )

```

Reply:
we could add a docstring, strange the linter doesn't fail